redisson/src/main/java/org/redisson/spring/session/RedissonSessionRepository.java

```diff
@@ -198,38 +201,40 @@ public boolean isExpired() {
         public String changeSessionId() {
             String oldId = delegate.getId();
             String id = delegate.changeSessionId();
-            if (redisson.getConfig().isClusterConfig()) {
-                Map<String, Object> oldState = map.readAllMap();
-                map.delete();
-                
-                map = redisson.getMap(keyPrefix + id, map.getCodec());
-                map.putAll(oldState);
-                
-                RBucket<String> bucket = redisson.getBucket(getExpiredKey(oldId));
-                long remainTTL = bucket.remainTimeToLive();
-                bucket.delete();
-                redisson.getBucket(getExpiredKey(id)).set("", remainTTL, TimeUnit.MILLISECONDS);
-            } else {
-                map.rename(keyPrefix + id);
-                redisson.getBucket(getExpiredKey(oldId)).rename(getExpiredKey(id));
-            }
+
+            Map<String, Object> oldState = map.readAllMap();
+            map.delete();
+
+            map = redisson.getMap(keyPrefix + id, map.getCodec());
+            map.putAll(oldState);
+
+            RBatch batch = redisson.createBatch(BatchOptions.defaults());
+            batch.getBucket(getExpiredKey(oldId)).remainTimeToLiveAsync();
+            batch.getBucket(getExpiredKey(oldId)).deleteAsync();
+
+            BatchResult<?> res = batch.execute();
+            List<?> list = res.getResponses();
+            Long remainTTL = (Long) list.get(0);
```

Comment:
@mrniko I am not sure about the casting to `Long` but without I am getting a type error "capture<?> is not type Long" ... Any feedback on this?

Reply:
That's fine.